app/src/app/api/v0/assistants/threads/messages/route.ts

```diff
@@ -0,0 +1,54 @@
+import { apiHandler } from "@/util/api";
+import { openai } from "@/util/openai";
+import { AssistantStream } from "openai/lib/AssistantStream";
+import { AssistantResponse } from "ai";
+
+const assistantId = process.env.OPENAI_ASSISTANT_ID as string;
+
+// Helper function for debugging
+// Extracting tokens, looking into events, ...
+const handleReadableStream = (stream: AssistantStream): AssistantStream => {
+  // stream.on("textCreated", () => console.log("Created"));
+  stream.on("textDelta", async (delta) => {
+    // Make sure token has annotation value
+    if (delta.annotations !== undefined && delta.annotations.length > 0) {
+      // console.log(delta.value);
+      // console.log(delta.annotations);
+      // Token can have only one annotation
+      const file_id = delta.annotations[0].file_citation.file_id;
```

Comment:
This line gives a type error in the CI pipeline (`npm run build`). If it runs but the type definition is wrong, we can ignore it if necessary, but let's make sure you're accessing this correctly...

Reply:
This is the `delta.annotations` object:

```[
  {
    index: 0,
    type: 'file_citation',
    text: '【6:0†source】',
    start_index: 928,
    end_index: 940,
    file_citation: { file_id: 'file-nI3Chhp90NOfufUM68jFsIkQ', quote: '' }
  }
]
```
So `file_citation.file_id` is part of it. I will add `// @ts-ignore`